Content.Server/Power/EntitySystems/BatterySystem.cs

```diff
@@ -145,6 +163,15 @@ public void SetCharge(EntityUid uid, float value, BatteryComponent? battery = nu
             RaiseLocalEvent(uid, ref ev);
         }
 
+        public void SetChargeCooldown(EntityUid uid, float value, BatterySelfRechargerComponent? batteryself = null)
+        {
+            if (!Resolve(uid, ref batteryself))
+                return;
+
+            var old = batteryself.AutoRechargeCooldown;
+            batteryself.AutoRechargeCooldown = Math.Clamp(value, 0, 9999);
+        }
```

Comment:
Please don't do magic clamps just set the value, also this should really be in shared.

Reply:
You are right, it should be in shared, but power is not in shared.

Reply:
Clamps removed, but I am not paid enough to move power to shared.